### DISCUSSION TITLE
I need help

### DISCUSSION THREAD

**Discussion by binbinlan** [neuralhydrology/neuralhydrology#83] [Q&A] [answered]
I want to use the example configuration In the neural hydrology config.yml.example, but the prompt filenotfounderror: [errno 2] no such file or directory: 'data \ \ \ \ 8_ basins_ dev_ set. Txt ', where can I configure this TXT file?

**Comment by binbinlan** [comment]
By the way ,I dont know how to write a config.yml.

**Comment by gauchm** [comment, marked as answer]
Hi @binbinlan,
the `8_basins_dev_set.txt` file is the file that specifies the basins used for training/evaluation. The path to that file is specified in the `config.yml` configuration file. Every neuralhydrology experiment is configured using such a .yml file.

The basin file is just a plain text file where each line contains the id of one basin. E.g., it could look like this:
```
01022500
01547700
02064000
03015500
```

In [lines 10-12 of the config.yml.example](https://github.com/neuralhydrology/neuralhydrology/blob/master/examples/config.yml.example#L10), you can see how the basin files for training, validation, and testing are configured as `./data/8_basins_dev_set.txt`. This means that neuralhydrology will look for the basin file in the subfolder `data/` (relative to your current working directory). You can either change that path in the config.yml, or place a basin file at the specified location.

You can edit the config.yml file in any text editor. In general, I'd recommend you start with the [neuralhydrology introduction tutorial](https://neuralhydrology.readthedocs.io/en/latest/tutorials/introduction.html) rather than the config.yml.example. Unlike the example config, the tutorials are configured to work out of the box and guide you through your first steps with neuralhydrology. You can find the code and configuration files to run the tutorials yourself [here](https://github.com/neuralhydrology/neuralhydrology/tree/master/examples/01-Introduction).

I hope this helps!

**Comment by mengziwen2000** [comment]
Hi, I would like to reproduce the method in Uncertainty estimation with DL for rainfall-runoff models, but I have not found a complete code to reproduce the process. I am a beginner and would like to understand the detailed principles of the method in detail by using the debugger method. Can you recommend the code, I would be grateful!

**Reply by danklotz** [reply]
HI mengziwen. That is entirely my fault. I started the Uncertainty paper with an old version of the code and never wrote an up-to-date tutorial on how to use the uncertainty estimation stuff. I am traveling for the next two weeks but will come back to you in Dec. Hope you can wait for so long :)

**Reply by mengziwen2000** [reply]
Thank you very much for your reply. Your tutorials are very helpful to us. I will keep following you. Good luck with your work.
